File: common/src/main/java/de/hype/bbsentials/shared/objects/json/ApiJsonList.java
package de.hype.bbsentials.shared.objects.json;

import com.google.gson.JsonArray;
import com.google.gson.JsonElement;

import java.util.ArrayList;
import java.util.List;

public class ApiJsonList {
    ApiJson parent;
    List<ApiJsonElement> values = new ArrayList<>();

    public ApiJsonList(ApiJson parent, String key) {
        this.parent = parent;
        JsonElement temp = parent.getJSONElementSafe();
        if (temp == null) return;
        JsonArray array = temp.getAsJsonObject().getAsJsonArray(key);
        if (array == null) return;
        for (JsonElement jsonElement : array) {
            values.add(new ApiJsonElement(jsonElement));
        }
    }

    public List<ApiJsonElement> toList() {
        return new ArrayList<>(values);
    }
}

File: common/src/main/kotlin/de/hype/bingonet/shared/objects/json/ApiJsonList.kt
package de.hype.bingonet.shared.objects.json

class ApiJsonList(var parent: ApiJson, key: String?) {
    var values: MutableList<ApiJsonElement?> = ArrayList()

    init {
        val temp = parent.jSONElementSafe
        if (temp != null) {
            val array = temp.getAsJsonObject().getAsJsonArray(key)
            if (array != null) {
                for (jsonElement in array) {
                    values.add(ApiJsonElement(jsonElement))
                }
            }
        }
    }

    fun toList(): MutableList<ApiJsonElement?> {
        return ArrayList<ApiJsonElement?>(values)
    }
}
